File: sample/src/main/java/ru/surfstudio/standard/app/dagger/ActiveActivityHolderModule.java
package ru.surfstudio.standard.app.dagger;

import dagger.Module;
import dagger.Provides;
import ru.surfstudio.android.core.app.dagger.scope.PerApplication;
import ru.surfstudio.android.core.util.ActiveActivityHolder;

@Module
public class ActiveActivityHolderModule {
    private ActiveActivityHolder activeActivityHolder;

    public ActiveActivityHolderModule(ActiveActivityHolder activeActivityHolder) {
        this.activeActivityHolder = activeActivityHolder;
    }

    @Provides
    @PerApplication
    ActiveActivityHolder provideActiveActivityHolder() {
        return activeActivityHolder;
    }
}

File: template/src/main/java/ru/surfstudio/standard/app/dagger/ActiveActivityHolderModule.kt
package ru.surfstudio.standard.app.dagger

import dagger.Module
import dagger.Provides
import ru.surfstudio.android.core.util.ActiveActivityHolder
import ru.surfstudio.android.dagger.scope.PerApplication

@Module
class ActiveActivityHolderModule(private val activeActivityHolder: ActiveActivityHolder) {

    @Provides
    @PerApplication
    internal fun provideActiveActivityHolder(): ActiveActivityHolder {
        return activeActivityHolder
    }
}
